### PR TITLE
feature/ldap-subsecond-timeouts

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Allow Subsecond timeouts for LDAP connections and randomize server queries

Impacts
=======
Admin and LDAP searching

Delete branch after merge
=========================
NO

NEWS file entries
=================

Enhancements
------------
* Allow Subsecond timeouts for LDAP connections
* Allow Randomization of the search order for a list of servers

### CODE REVIEW

File: lib/pf/Authentication/Source/LDAPSource.pm [hunk 1]
```diff
@@ -51,9 +51,9 @@ Readonly our %ATTRIBUTES_MAP => (
 has '+type' => (default => 'LDAP');
 has 'host' => (isa => 'Maybe[Str]', is => 'rw', default => '127.0.0.1');
 has 'port' => (isa => 'Maybe[Int]', is => 'rw', default => 389);
-has 'connection_timeout' => ( isa     => 'Int', is => 'rw', default => 5 );
-has 'write_timeout' => (isa => 'Int', is => 'rw', default => $DEFAULT_LDAP_WRITE_TIMEOUT);
-has 'read_timeout' => (isa => 'Int', is => 'rw', default => $DEFAULT_LDAP_READ_TIMEOUT);
+has 'connection_timeout' => ( isa     => 'Num', is => 'rw', default => 5 );
```
Comment (fdurand) [inline lib/pf/Authentication/Source/LDAPSource.pm]: Just lower a bit the value, 5 is really high

Comment (jrouzierinverse) [reply lib/pf/Authentication/Source/LDAPSource.pm]: What do you recommend? I was thinking 1 second or 0.5?

Comment (fdurand) [reply lib/pf/Authentication/Source/LDAPSource.pm]: Let's say 1 for now.

**Comment by jrouzierinverse** [comment]
done

**Pull request merged**